cloud/services/securitygroups/client.go

```diff
@@ -58,11 +57,22 @@ func (ac *AzureClient) Get(ctx context.Context, resourceGroupName, sgName string
 }
 
 // CreateOrUpdate creates or updates a network security group in the specified resource group.
-func (ac *AzureClient) CreateOrUpdate(ctx context.Context, resourceGroupName string, sgName string, sg network.SecurityGroup) error {
-	future, err := ac.securitygroups.CreateOrUpdate(ctx, resourceGroupName, sgName, sg)
+func (ac *AzureClient) CreateOrUpdate(ctx context.Context, resourceGroupName string, sgName string, sg network.SecurityGroup, etag string) error {
```

Comment:
Etag is on `network.SecurityGroup`. Perhaps, rather than adding it to the signature of the func, make a copy of the etag field before calling `ac.securitygroups.CreateOrUpdatePreparer`, then add the copy to the header.

I believe `ac.securitygroups.CreateOrUpdatePreparer` will set the `Etag *string` ptr to nil b/c if it were to be included in the body of the PUT, the service would reject the request with a 400 error.

Reply:
Updated the PR